### ISSUE TITLE
buttonRoot isn't set for edit frame

### ISSUE THREAD

**Issue by pmandava** [mikeedwards83/Glass.Mapper#248]
Hi All, 

I am working on Sitecore 8.2 (rev. 160729) with glass 4.2.1.188.
When trying to use EditFrames , I run into "buttonRoot isn't set for edit frame", when already logged into sitecore.

And multiple Edit Frame`s are created in core db.
Please find the attached for reference.

Is this already a known issue ?

Thanks,
-Purnima.
![2016-10-06_12-47-22](https://cloud.githubusercontent.com/assets/14077620/19161816/54bf5316-8bc3-11e6-9d47-57a5f5eef540.jpg)
![2016-10-06_12-47-40](https://cloud.githubusercontent.com/assets/14077620/19161814/54ba72ec-8bc3-11e6-86b2-89b623f822c3.jpg)

**Comment by artur-yakimovich** [comment]
Hi, Purnima.
Have you solved this issue? Looks like I have the same one.

Thanks, 
Artur

**Comment by artur-yakimovich** [comment]
After deep investigation I've found out, that the issue is related to Sitecore 8.2 API - GetItem(string path) method:

For example,
![image](https://cloud.githubusercontent.com/assets/17310961/20665159/729d2144-b56f-11e6-87e9-a6cbe958ba3a.png)

But,
![image](https://cloud.githubusercontent.com/assets/17310961/20665170/863c9f72-b56f-11e6-8959-6b81dbce0c79.png)

Both items exist in Sitecore:
![image](https://cloud.githubusercontent.com/assets/17310961/20665181/9519029c-b56f-11e6-9864-2689560ed5c9.png)

It happens only for items with GUID names. 
Issue is not reproduced for Sitecore 8.0 and 8.1

**Comment by mikeedwards83** [comment]
Hum, interesting. Thanks for investigating this. Maybe it is a problem with
hyphens?

**Comment by mikeedwards83** [comment]
Locally I have test getting an item with a guid name and it works fine. I am not sure how you are getting edit frame folders with Guids, the code we use shouldn't generated this folder name.

**Comment by artur-yakimovich** [comment]
@mikeedwards83,

Hm, I'm a bit confused with your last answer.

Could you please look at the code from line 59 in [source](https://github.com/mikeedwards83/Glass.Mapper/blob/master/Source/Glass.Mapper.Sc/Pipelines/GetChromeData/EditFrameBuilder.cs)?

This code creates new item with "GUID" name in case item does not exist in database.

**Comment by artur-yakimovich** [comment]
@mikeedwards83,
Sent an email with details on hello@glass.lu

Thanks and best regards

**Comment by tamalik** [comment]
I have noticed a similar issue with Sitecore 8 initial update. This looks like a bug in Sitecore API where it fails to return the data from a Database if the Item name matches a Guid value like E1C403F7-A15C-4FE5-976F-9FB5437AB207. I tried in our project in for both Core and Master Database and it depicts the same behavior. If characters in Item is reduced by one or two so that it doesn't match Guid value then it returns the Item. I am hoping that Sitecore will accept this bug in Factory.GetDatabase().GetItem(strPath) API.
